core/src/main/java/org/pentaho/platform/web/http/api/resources/SchedulerOutputPathResolver.java

```diff
@@ -96,11 +102,43 @@ public String getJobName() {
     return scheduleRequest.getJobName();
   }
 
-  public String getActionUser() {
-    return pentahoSession.getName();
+  private String getScheduleOwner() {
+    return scheduleOwner;
+  }
+
+  /**
+   * Gets the job schedule owner using the job parameter `ActionAdapterQuartzJob-ActionUser`.
+   *
+   * @return A trimmed non-empty string containing the value of the job parameter or null.
+   */
+  private String getScheduleOwnerFromRequest() {
+    String userName = scheduleRequest.getJobParameters().stream()
+      .filter( jobParameter -> StringUtils.equals( ( jobParameter ).getName(), IScheduler.RESERVEDMAPKEY_ACTIONUSER ) )
+      .findFirst()
+      .map( jobParameter -> jobParameter.getValue() != null ? ( (String) jobParameter.getValue() ).trim() : null )
+      .orElse( null );
+
+    // Schedule Owner should have already been validated with a fallback to administrator
+    // if there is a problem with the schedule owner at this point, it is a bigger issue.
+    if ( userName == null || userName.isEmpty() ) {
```

Comment:
Are there any test scenarios catching/asserting `IllegalArgumentException` in SchedulerOutputPathResolverTest.java:

- userName == null
- username.isEmpty()

Reply:
updated UTs  with some refactoring of repeated code and added coverage, more notes in commit message
https://github.com/pentaho/pentaho-scheduler-plugin/pull/242/commits/dc1e0c05aa778e20d7241093392be3e4e3ee4148